File: python/examples/functions/bubble_sort.py

def bubble_sort(*values):
    values = list(values)
    for ix in range(len(values)-1):
        for jx in range(len(values)-1-ix):
            if values[jx] > values[jx+1]:
                values[jx], values[jx+1] = values[jx+1], values[jx]
    return values

print(bubble_sort(1, 2, 3))
print(bubble_sort(3, 2, 1))
print(bubble_sort(10, 9, 8, 7, 6, 5, 4, 3, 2, 1))

File: books/perl/src/examples/other/bubble_sort.pl
use strict;
use warnings;
use 5.010;
use Benchmark qw(:hireswallclock);

my @numbers = (1 .. 4);
print "@numbers\n";
my @sorted = bubble_sort(@numbers);
print "@sorted\n";

#timethese(1, {
#    '1000'  => sub { bubble_sort(1..1000) },
#    '10000' => sub { bubble_sort(1..10000) },
#});

#say(timeit(10, sub { bubble_sort(1..1000)  })->real);

sub bubble_sort {
    my @items = @_;
    for my $ix (0 .. $#items) {
        for my $jx (0 .. $#items-$ix-1) {
            if ($items[$jx] < $items[$jx+1]) {
                ($items[$jx], $items[$jx+1]) = ($items[$jx+1], $items[$jx]);
                print "@items\n";
            }
        }
    }
    return @items;
}
